test/test_prototype_builtin_datasets.py

```diff
@@ -25,9 +25,10 @@ def extract_datapipes(dp):
     return get_all_graph_pipes(traverse(dp, only_datapipe=True))
 
 
-@pytest.fixture
+@pytest.fixture(autouse=True)
 def test_home(mocker, tmp_path):
     mocker.patch("torchvision.prototype.datasets._api.home", return_value=str(tmp_path))
+    mocker.patch("torchvision.prototype.datasets.home", return_value=str(tmp_path))
```

Comment:
We need this extra patch since the mock data generation is calling the public interface `datasets.home()` whereas the loading does not:

https://github.com/pytorch/vision/blob/40a0ab797945e20fda0b35030094fb6741301e07/torchvision/prototype/datasets/_api.py#L4

Reply:
Q: I'm always incredibly confused by the behavior of `mocker.patch`.

Is it because it patches names (references to objects) instead of patching objects?

Reply:
It patches attributes. Not sure if that is what you mean by "names". The above is roughly equivalent to

```py
torchvision.prototype.datasets._api.home = lambda *_, **_: str(tmp_path)
torchvision.prototype.datasets.home = lambda *_, **_: str(tmp_path)
```

Thus, it patches the attribute `home` on the two modules rather than the actual function.